command/run.go

```diff
@@ -144,6 +145,41 @@ func (c *RunCommand) Synopsis() string {
 
 // Run executes the command.
 func (c *RunCommand) Run(args []string) int {
+	// a randomly-named temporary directory
+	var tmp string
+	var cleanup func(hclog.Logger)
+	var err error
+
+	// Default teeWriter just prints to stdout
+	var teeWriter io.Writer = os.Stdout
+	l := configureLogging("hcdiag", teeWriter)
+
+	// Create a temporary directory and logfile
+	if !c.dryrun {
+		if tmp, cleanup, err = util.CreateTemp(c.destination); err != nil {
+			fmt.Println("Failed to create temp directory. error:", err)
+			return SetupError
+		}
+		// remove the temporary directory and its contents
+		defer cleanup(l)
+
+		// Set up stdout/logfile output
+		logfile, err := os.Create(path.Join(tmp, "hcdiag.log")) // creates a file at current directory
```

Comment:
Would `filepath.Join` be safer here for Windows machines?

Reply:
D'oh! Yes, thank you. Made myself a nice windows bug in another place where I mixed up path/filepath.